src/Mod/Draft/DraftGeomUtils.py

```diff
@@ -1116,9 +1116,18 @@ def offset(edge,vector,trim=False):
 
 def isReallyClosed(wire):
     "checks if a wire is really closed"
-    if len(wire.Edges) == len(wire.Vertexes): return True
-    v1 = wire.Vertexes[0].Point
-    v2 = wire.Vertexes[-1].Point
+
+    ## TODO yet to find out why not use wire.isClosed() direct, in isReallyClosed(wire)
+
+    # Remark out below - Found not true if a vertex is used again in a wire in sketch ( e.g. wire with shape like 'd', 'b', 'g'... )
+    #if len(wire.Edges) == len(wire.Vertexes): return True
+
+    # Found cases where Wire[-1] are not 'last' vertexes (e.g. Part.Wire( Part.__sortEdges__( <Rectangle Geometries>.toShape() ) )
+    # aboveWire.isClosed() == True, but Wire[-1] are the 3rd vertex for the rectangle
+    # - use Edges[i].Vertexes[0/1] instead
+    length = len(wire.Edges)
+    v1 = wire.Edges[0].Vertexes[0].Point					#v1 = wire.Vertexes[0].Point
```

Comment:
Please don't add tabs to the code. Use spaces. For this comment, it should be a two space separation from the code, so that it is compliant to Python styleguides.
```
v1 = wire.Edges[0].Vertexes[0].Point  # v1 = wire.Vertexes[0].Point
```

Reply:
Thanks, I have a look at https://www.python.org/dev/peps/pep-0008/#inline-comments which recommend 2 spaces separation indeed.  It shows examples that inline comments are separated consistently at a fixed margin.

I just learn to use gedit to edit the code, it happens to offer a 'Dispaly right margin' option and making inline comments all at this location - same as the above website recommends -  consistent and readable.

Using Tab after the main code seem most handy to make these inline comments consistent in location (though afterall inline comments in the styleguide should be used sparingly).  Need more works to keep these align when main code are edited.  Will try if there is quicker way to do it.

Reply:
Avoid inline comments. And if you add them, don't try to make them aligned; just two spaces is enough. Use the Spyder editor to write Python code. It will highlight places that can be improved.